string pid
string verb
string location
string tool
string[] ingredients